src/main/java/games/strategy/engine/ClientFileSystemHelper.java

```diff
@@ -32,55 +30,46 @@ private ClientFileSystemHelper() {}
   /**
    * @return Folder that is the 'root' of the tripleA binary installation. This folder and
    *         contents contains the versioned content downloaded and initially installed. This is
-   *         in contrast to the user root folder that is not replaced between between installations.
+   *         in contrast to the user root folder that is not replaced between installations.
    */
   public static File getRootFolder() {
-    final String fileName = getGameRunnerFileLocation(GameRunner.class.getSimpleName() + ".class");
+    final String classFilePath = getGameRunnerClassFilePath();
 
-    final String tripleaJarName = "triplea.jar!";
-    if (fileName.contains(tripleaJarName)) {
-      return getRootFolderRelativeToJar(fileName, tripleaJarName);
+    final String jarFileName = String.format("triplea-%s-all.jar!", ClientContext.engineVersion().getExactVersion());
+    if (classFilePath.contains(jarFileName)) {
+      return getRootFolderRelativeToJar(classFilePath, jarFileName);
     }
 
-    final String tripleaJarNameWithEngineVersion = getTripleaJarWithEngineVersionStringPath();
-    if (fileName.contains(tripleaJarNameWithEngineVersion)) {
-      return getRootFolderRelativeToJar(fileName, tripleaJarNameWithEngineVersion);
-    }
-
-    return getRootRelativeToClassFile(fileName);
+    return getRootFolderRelativeToClassFile(classFilePath);
   }
 
-  private static String getGameRunnerFileLocation(final String runnerClassName) {
+  private static String getGameRunnerClassFilePath() {
+    final String runnerClassName = GameRunner.class.getSimpleName() + ".class";
     final URL url = GameRunner.class.getResource(runnerClassName);
-    String fileName = url.getFile();
+    String path = url.getFile();
 
     try {
       // Deal with spaces in the file name which would be url encoded
-      fileName = URLDecoder.decode(fileName, StandardCharsets.UTF_8.name());
+      path = URLDecoder.decode(path, StandardCharsets.UTF_8.name());
```

Comment:
Since this method is only used internally, couldn't this method just return a URL or URI object instead of returning a specially formatted String?
Also why is this method even necessary?
I believe `Someclass.class.getResource("/")` would get the root directory without any intermediate steps. And then all we need to check is wether this URL points to a file (the triplea.jar) or a directory (the compiled root)

Reply:
Probably, but that should be accompanied by a good set of unit tests because there appears to be a few subtle issues in how the value returned by this method is subsequently processed by downstream code.  For example, there is code in `getRootFolderRelativeToClassFile()` that assumes the `GameRunner` class file was used in the input path.  That should be addressed, as well.

I think that's a complex enough refactoring that it justifies a separate PR.

**NOTE:** I did simplify this code a bit more by treating the `UnsupportedEncodingException` as an impossible condition.